src/components/AppliedFilters.tsx

```diff
@@ -58,42 +58,29 @@ export interface AppliedFiltersProps {
  * @returns A React element for the applied filters
  */
 export function AppliedFilters(props: AppliedFiltersProps): JSX.Element {
-  const nlpFilters = useSearchState(state => state.vertical.appliedQueryFilters);
   const isLoading = useSearchState(state => state.searchStatus.isLoading);
-  const hasResults = !!useSearchState(state => state.vertical.results);
-  const filters = useStateUpdatedOnSearch(state => state.filters);
 
   const {
-    hiddenFields,
+    hiddenFields = DEFUALT_HIDDEN_FIELDS,
     customCssClasses = {},
     hierarchicalFacetsDelimiter = DEFAULT_HIERARCHICAL_DELIMITER,
     hierarchicalFacetsFieldIds
   } = props;
 
-  const appliedFilters: GroupedFilters = useMemo(() => {
-    return pruneAppliedFilters(
-      hasResults ? (filters ?? {}) : {},
-      nlpFilters ?? [],
-      hiddenFields ?? ['builtin.entityType'],
-      hierarchicalFacetsFieldIds ?? [],
-      hierarchicalFacetsDelimiter
-    );
-  }, [
-    hasResults,
-    filters,
-    hiddenFields,
-    hierarchicalFacetsDelimiter,
-    hierarchicalFacetsFieldIds,
-    nlpFilters
-  ]);
+  const removableFilters = useRemovableFilters(
+    hierarchicalFacetsFieldIds, hierarchicalFacetsDelimiter, hiddenFields);
+  const nlpFilterDisplayNames = useNlpFilterDisplayNames(removableFilters.map(f => f.filter), hiddenFields);
```

Comment:
It feels a bit like we're making these functions Hooks out of necessity. A Hook should allow us to reuse stateful logic between Components. We don't invoke these methods anywhere else, right? They use Hooks internally, so they have to be Hooks at the top-level. 

Instead, what do you think about making these just normal functions? Any memo-izing would happen in this file. Any `answersActions` or `answersState` dependencies would be injected.

Reply:
I like how making these hooks lets us call hooks from within them. This results in more concise code. I think whether or not we invoke these methods in other components doesn't effect whether or not these functions are hooks or not - by that same logic regular utility functions are also meant to be shared between components. I would actually argue that because these functions aren't shared between components that it's a little simpler to just make these hooks and not worry about passing parameters down to these functions. Having to pass tons of parameters down was something I really disliked about appliedfilterutils, especially because those parameters were only going to be used within those utility functions and not anywhere else in the component.

Reply:
I'm definitely open to turning these into regular functions if we need to share them with other components, though! I think doing that refactor isn't something that will get harder with time, so there's no need to jump start it before it's needed.